### ISSUE TITLE
Convert saved model - issue generated shapes

### ISSUE THREAD

**Issue by hahmad2008** [tensorflow/tensorflow#42153]
**System information**
- OS: MAC
- TensorFlow version 2.4.0-dev20200805

**Command used to run the converter or code if you’re using the Python API**

```
!pip install tf-nightly

import tensorflow as tf
converter = tf.lite.TFLiteConverter.from_saved_model('ssd_mobilenet_v2_320x320_coco17_tpu-8/saved_model')
converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.experimental_new_converter = True

converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS, tf.lite.OpsSet.SELECT_TF_OPS]
tflite_model = converter.convert()

open("m.tflite", "wb").write(tflite_model)
```

**Original Model Input & Output shape**

![image](https://user-images.githubusercontent.com/68266028/89719024-c337ef00-d9cc-11ea-8858-afc169c5237c.png)

**Converted Model Input & Output shape**

![image](https://user-images.githubusercontent.com/68266028/89719032-e367ae00-d9cc-11ea-9ff2-f5700e7c36a7.png)

**Also, please include a link to the saved model {{[LINK MODEL](http://download.tensorflow.org/models/object_detection/tf2/20200711/ssd_resnet50_v1_fpn_640x640_coco17_tpu-8.tar.gz)}}**

**Please let me know if I have missed anything here, I feel that the model has something wrong! (Shapes are not matched)  As I replaced the model in the Android version of object detection, it gives me errors.**

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with the latest TF-nightly. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/164b81ed8effbeafc573b41ded1a64c6/42153.ipynb). Thanks!

**Comment by hahmad2008** [comment]
@amahendrakar So there is a problem with mismatching the input-output shapes, right?

**Comment by Lizhi-Liao** [comment]
Meet the same problem, I tried to convert an LSTM model with input (None, 200), but after the convert, the input shape for the tflite model becomes (1,1)

original model codes:
inputs = keras.Input(shape=(None,), dtype="int32")
# Embed each integer in a 128-dimensional vector
x = layers.Embedding(max_features, 128)(inputs)
# Add 2 LSTMs
x = layers.LSTM(64, return_sequences=True)(x)
x = layers.LSTM(64)(x)           
# Add a classifier
outputs = layers.Dense(1, activation="sigmoid")(x)
model = keras.Model(inputs, outputs)
model.summary()

![image](https://user-images.githubusercontent.com/23717722/90100790-66f60780-dd0b-11ea-89db-2733a3cd23b5.png)

**Comment by renjie-liu** [comment]
Current tflite has limited dynamic shape support.

You can either set a fixed shape or calling resizeInputShape

**Comment by Paranormaly** [comment]
I have the same issue trying to convert custom trained ssd_mobilenet_v1_fpn_640x640_coco17_tpu-8 
Any way around?

**Comment by krn-sharma** [comment]
@Paranormaly  I am also facing exact same issue. Have you resolved it?

**Comment by QianZixi** [comment]
I am also facing the problem.
I have modified the data in the code as much as possible

**Comment by DeepAlex1247** [comment]
Having the same issue with an Efficientdet_d0_512x512 model that I converted to tflite with tf-nightly 2.5.00
input shape is set to [1, 1, 1, 3]

**Comment by ricardoantonello** [comment]
Have the same issue with ssdmobilenetv2 in tf2.4 and '2.5.0-dev20210114'

**Comment by jennya** [comment]
I am also having this issue - does anyone have a resolution?

**Comment by ricardoantonello** [comment]
> I am also having this issue - does anyone have a resolution?

Hi,
I finally solve the issue using export_tflite_graph_tf2.py from OD API.
I used de checkpoints i have trainning before, and the instructions here:
python object_detection/export_tflite_graph_tf2.py
--pipeline_config_path path/to/ssd_model/pipeline.config
--trained_checkpoint_dir path/to/ssd_model/checkpoint
--output_directory path/to/exported_model_directory

pipeline_config_path is the same config file i used to trainning.
My version: TF 2.4 Gpu

Good luck!

**Comment by Marouaneghoulami** [comment]
> > I am also having this issue - does anyone have a resolution?
> 
> Hi,
> I finally solve the issue using export_tflite_graph_tf2.py from OD API.
> I used de checkpoints i have trainning before, and the instructions here:
> python object_detection/export_tflite_graph_tf2.py
> --pipeline_config_path path/to/ssd_model/pipeline.config
> --trained_checkpoint_dir path/to/ssd_model/checkpoint
> --output_directory path/to/exported_model_directory
> 
> pipeline_config_path is the same config file i used to trainning.
> My version: TF 2.4 Gpu
> 
> Good luck!

It worked for SSD Mobilenet v2.

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this issue, as you are using an older version of tensorflow(1.x) which is officially considered as end of life. We recommend that you upgrade to 2.4 or later version and let us know if the issue still persists in newer versions. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, Please open a new issue for any help you need against 2.x, and we will get you the right help.

**Comment by sachinprasadhs** [comment]
@hahmad2008 , Hi did the below comment help you resolve your issue?

> > I am also having this issue - does anyone have a resolution?
> 
> Hi,
> I finally solve the issue using export_tflite_graph_tf2.py from OD API.
> I used de checkpoints i have trainning before, and the instructions here:
> python object_detection/export_tflite_graph_tf2.py
> --pipeline_config_path path/to/ssd_model/pipeline.config
> --trained_checkpoint_dir path/to/ssd_model/checkpoint
> --output_directory path/to/exported_model_directory
> 
> pipeline_config_path is the same config file i used to trainning.
> My version: TF 2.4 Gpu
> 
> Good luck!

**Comment by rokaijano** [comment]
I encountered the same problem. For me, the problem was that I used **tf.saved_model.save()** function to save my model ( which was a custom keras model ), using the **Model.save()** function solved the problem.
